File: app/src/main/java/com/kickstarter/ui/viewholders/DiscoveryOnboardingViewHolder.java
package com.kickstarter.ui.viewholders;

import android.view.View;
import android.widget.Button;

import com.kickstarter.R;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import butterknife.Bind;
import butterknife.ButterKnife;
import butterknife.OnClick;

public final class DiscoveryOnboardingViewHolder extends KSViewHolder {
  protected @Bind(R.id.login_tout_button) Button lgoinToutButton;

  private final Delegate delegate;
  public interface Delegate {
    void discoveryOnboardingViewHolderLoginToutClick(DiscoveryOnboardingViewHolder viewHolder);
  }

  public DiscoveryOnboardingViewHolder(final @NonNull View view, final @NonNull Delegate delegate) {
    super(view);
    this.delegate = delegate;
    ButterKnife.bind(this, view);
  }

  @Override
  public void bindData(final @Nullable Object data) throws Exception {}

  public void onBind() {}

  @OnClick(R.id.login_tout_button)
  protected void loginToutClick() {
    this.delegate.discoveryOnboardingViewHolderLoginToutClick(this);
  }
}

File: app/src/main/java/com/kickstarter/ui/viewholders/DiscoveryOnboardingViewHolder.kt
package com.kickstarter.ui.viewholders

import com.kickstarter.databinding.DiscoveryOnboardingViewBinding

class DiscoveryOnboardingViewHolder(
    private val binding: DiscoveryOnboardingViewBinding,
    private val delegate: Delegate
) : KSViewHolder(binding.root) {

    interface Delegate {
        fun discoveryOnboardingViewHolderLoginToutClick(viewHolder: DiscoveryOnboardingViewHolder?)
    }

    override fun onBind() {
        binding.loginToutButton.setOnClickListener {
            loginToutClick()
        }
    }

    override fun bindData(data: Any?) {
    }

    private fun loginToutClick() {
        delegate.discoveryOnboardingViewHolderLoginToutClick(this)
    }
}
